PowerPC32: #error "TARGET_RUST_OS_INFO already defined in linux.h (rs6000)

Trying to build revision 325ef69b132819b824ae757695d9724e503f7256 on 32-bit PowerPC fails with:

```
g++ -c   -g -O2 -DIN_GCC     -fno-exceptions -fno-rtti -fasynchronous-unwind-tables -W -Wall -Wno-narrowing -Wwrite-strings -Wcast-qual -Wno-error=format-diag -Wmissing-format-attribute -Woverloaded-virtual -pedantic -Wno-long-long -Wno-variadic-macros -Wno-overlength-strings -fno-common  -DHAVE_CONFIG_H  -DGENERATOR_FILE -fno-PIE -I. -Ibuild -I../../gcc -I../../gcc/build -I../../gcc/../include  -I../../gcc/../libcpp/include  \
        -o build/genpreds.o ../../gcc/genpreds.c
In file included from ./tm.h:32,
                 from ../../gcc/genpreds.c:26:
../../gcc/config/rs6000/linux.h:63:3: error: #error "TARGET_RUST_OS_INFO already defined in linux.h (rs6000) - c++ undefines it and redefines it."
   63 | # error "TARGET_RUST_OS_INFO already defined in linux.h (rs6000) - c++ undefines it and redefines it."
      |   ^~~~~
../../gcc/config/rs6000/linux.h:65: warning: "TARGET_RUST_OS_INFO" redefined
   65 | #define TARGET_RUST_OS_INFO()          \
      |
In file included from ./tm.h:31,
                 from ../../gcc/genpreds.c:26:
../../gcc/config/rs6000/sysv4.h:554: note: this is the location of the previous definition
  554 | #define TARGET_RUST_OS_INFO()  \
      |
make[2]: *** [Makefile:2833: build/genpreds.o] Error 1
make[2]: Leaving directory '/home/glaubitz/gccrs/build/gcc'
make[1]: *** [Makefile:4415: all-gcc] Error 2
make[1]: Leaving directory '/home/glaubitz/gccrs/build'
make: *** [Makefile:944: all] Error 2
```

--- a/gcc/config/rs6000/linux.h
+++ b/gcc/config/rs6000/linux.h
@@ -59,9 +59,8 @@
     }							\
   while (0)
 
-#ifdef TARGET_RUST_OS_INFO
-# error "TARGET_RUST_OS_INFO already defined in linux.h (rs6000) - c++ undefines it and redefines it."
-#endif
+/* Override less-specific rs6000/sysv4.h definition. */
+#undef TARGET_RUST_OS_INFO
 #define TARGET_RUST_OS_INFO()          \
   do {                                 \
     GNU_USER_TARGET_RUST_OS_INFO();		\

